aten/src/ATen/function_wrapper.py

```diff
@@ -779,15 +779,12 @@ def gen_device_init(option, backend_type_env):
     # type: (FunctionOption, Environment) -> List[str]
     # generate a device init statement, if the passed function option is a Tensor factory.
     #
-    if not is_factory(option):
-        return []
-
-    name = option['name']
-    device_type = backend_type_env['DeviceType']
-    assert device_type in ['CPU', 'CUDA'], \
-        "{}: unsupported device type '{}'".format(name, device_type)
-    return ['globalLegacyTypeDispatch().init{}();'.format(device_type)]
-
+    if is_factory(option):
+        name = option['name']
+        device_type = backend_type_env['DeviceType']
+        if device_type == 'CUDA' or device_type == 'HIP':
+            return ['globalContext().lazyInit{}();'.format(device_type)]
```

Comment:
Technically, this is still virtual, but this is a much appreciated simplification!

Reply:
Oh, so true :) I mostly decided to stop drilling at the Context API bc I saw other cuda-specific code calling into it, and the hooks stuff behind it seemed complicated. But so yeah, "devirtualize" in the sense of "make one step less virtual" 😬 ...is there a straightforward way to take it even further?